cf_xarray/accessor.py

```diff
@@ -170,6 +180,69 @@ def _get_groupby_time_accessor(var: Union[DataArray, Dataset], key: str) -> List
         return []
 
 
+def _get_custom_criteria(
+    obj: Union[DataArray, Dataset], key: str, criteria=None
+) -> List[str]:
+    """
+    Translate from axis, coord, or custom name to variable name optionally
+    using ``custom_criteria``
+
+    Parameters
+    ----------
+    obj : DataArray, Dataset
+        DataArray belonging to the coordinate to be checked
+    key : str, ["X", "Y", "Z", "T", "longitude", "latitude", "vertical", "time"], or a key from user-defined custom_criteria
+        key to check for.
+    criteria : dict, optional
+        Criteria to use to map from variable to attributes describing the
+        variable. An example is coordinate_criteria which maps coordinates to
+        their attributes and attribute values. If user has defined
+        custom_criteria, this will be used by default.
+
+    Returns
+    -------
+    List[str], Variable name(s) in parent xarray object that matches axis, coordinate, or custom `key`
+
+    """
+
+    if isinstance(obj, DataArray):
+        obj = obj._to_temp_dataset()
+
+    if criteria is None:
+        if not OPTIONS["custom_criteria"]:
+            return []
+        criteria = OPTIONS["custom_criteria"]
+
+    if criteria is not None:
+        criteria = always_iterable(criteria, allowed=(tuple, list, set))
+
+    criteria = ChainMap(*criteria)
+
+    results: Set = set()
+    if key in criteria:
+        for criterion, expected in criteria[key].items():
+            for var in obj.variables:
+                # Treat expected as regex
+                expected_vals = [*expected.values()][0]
```

Comment:
Can we not just use `kind.values()` instead?

Reply:
This leads to an error in the following regex line (variable still called "expected" instead of "kind"). 

![image](https://user-images.githubusercontent.com/3487237/122986698-61267500-d365-11eb-8bac-7e59b0904988.png)

because `expected_vals` is not just a list or tuple:
```
ipdb>  expected_vals
dict_values([('sea_surface_elev*',)])
```

So I have that particular syntax in there to try to work around the dictionary structure since the ChainMap approach didn't work in Python 3.7 or 3.8

Reply:
So I didn't change this other than the variable name change you suggested.